packages/cli/lib/ModuleManager.ts

```
@@ -0,0 +1,221 @@
+    for (const plugins of this.plugins.values()) {
+      for (const plugin of plugins.values()) {
+        if (plugin.getOptions) {
+          const options = await plugin.getOptions();
+
+          for (const option of options.keys()) {
+            yargs = yargs.option(`${plugin.name}-${option}`, {
+              ...options.get(option),
+              group: `${plugin.name} ${options.get(option).group || ""}`,
+            });
+          }
+        }
+      }
+    }
+    return yargs;
+  }
+
+  async getModulePath(module: string): Promise<string> {
+    let modulePath = "";
+
+    if (module.startsWith("file:")) {
+      // It is a file path
+      modulePath = path.resolve(module.replace("file:", ""));
+      if (!shell.test("-e", modulePath)) {
+        throw new Error(`Filepath does not lead to an module, path: ${module}`);
+      }
+    } else {
+      // It is a npm package
+      modulePath = path.resolve(path.join("node_modules", module));
+
+      if (!shell.test("-e", modulePath)) {
+        // it is not locally installed lets try global
+        modulePath = path.resolve(path.join(globalModules, module));
+      }
+
+      if (!shell.test("-e", modulePath)) {
+        // it is not installed locally nor globally
+        // TODO maybe auto install?
+        throw new Error(
+          `Package is not installed locally or globally, package: ${module}`
+        );
+      }
+    }
+
+    return modulePath;
+  }
+
+  async loadApplication(application: string) {
+    const modulePath = await this.getModulePath(application);
+    const { syntestApplication } = await import(modulePath);
+    const applicationInstance: ApplicationInterface =
+      new syntestApplication.default();
+
+    // check requirements
+    if (!applicationInstance.getCommandModules) {
+      throw new Error("SynTest application module required command property!");
+    }
+
+    this._applications.set(application, applicationInstance);
```

What if this already exists? We should throw an error here

---

fixed.